src/Reflection/ReflectionClass.php

```diff
@@ -490,6 +495,60 @@ public function hasConstant(string $name) : bool
     }
 
     /**
+     * Get the reflection object of the specified class constant.
+     *
+     * Returns null if not specified.
+     *
+     * @param string $name
+     * @return ReflectionClassConstant|null
+     */
+    public function getReflectionConstant(string $name) : ?ReflectionClassConstant
+    {
+        $constants = $this->getReflectionConstants();
+
+        if (!array_key_exists($name, $constants)) {
+            return null;
+        }
+
+        return $constants[$name];
+    }
+
+    /**
+     * Returns array of constants for this class
+     *
+     * @param array $accumulator
+     * @param Node\Stmt $stmt
+     * @return array
+     */
+    private function processReflectionConstants(array $accumulator, Node\Stmt $stmt) : array
+    {
+        if ($stmt instanceof ClassConst) {
+            $const = ReflectionClassConstant::createFromNode($this->reflector, $stmt, $this);
+            $accumulator[$const->getName()] = $const;
+        }
+
+        return $accumulator;
+    }
+
+    /**
+     * Get an array reflection object of the defined constants in this class.
+     *
+     * @return ReflectionClassConstant[]
+     */
+    public function getReflectionConstants() : array
+    {
+        if (null === $this->cachedReflectionConstants) {
+            $this->cachedReflectionConstants = array_reduce(
```

Comment:
@Ocramius native reflection not provide access for constants inherited from parents. Current behavior is exactly the same as in the native reflection.
```
class A
{
    const A = 1;
}
class B
{
    const B = 2;
}
foreach ([A::class, B::class] as $class) {
    echo 'Native ', $class, ':', PHP_EOL;
    $ref = new ReflectionClass($class);
    foreach ($ref->getReflectionConstants() as $const) {
        echo $const;
    }

    echo 'Better ', $class, ':', PHP_EOL;
    $ref = $reflector->reflect($class);
    foreach ($ref->getReflectionConstants() as $const) {
        echo $const;
    }
}
```
Output:
```
Native A:
Constant [ public integer A ] { 1 }
Better A:
Constant [ public integer A ] { 1 }
Native B:
Constant [ public integer B ] { 2 }
Better B:
Constant [ public integer B ] { 2 }
```

And traits doesn't support constants.